cmake/checks/check_02_system_features.cmake

```diff
@@ -105,6 +105,20 @@ ENDIF()
 #                                                                      #
 ########################################################################
 
+#
+# Put an end to user's suffering from cygwin's defects
+#
+IF( CMAKE_SYSTEM_NAME MATCHES "CYGWIN" OR
+    CMAKE_SYSTEM_NAME MATCHES "Windows" )
+  IF(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    MESSAGE(FATAL_ERROR
+      "\nCygwin and forks such as MinGW and MinGW-64 are unsupported due to "
+      "multiple unresolved miscompilation issues.\nIf you still want to "
+      "try to compile deal.II, find this check and remove it. You have been "
```

Comment:
> I would just say "use MSVC instead".

Nope. As long as I get a compilation error for current HEAD and the MSVC
version I have installed on Windows, I won't do that ;-)